### PR TITLE
feat(status-api): main app setup

### PR DESCRIPTION
**Pull request by joeldavidw**
#### Which issue(s) does this PR fixes?:

Part of #1270

### CODE REVIEW

File: apps/status-api/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+As part of [#580](https://github.com/DeFiCh/jellyfish/issues/580) consolidation efforts. We had multiple projects that
+were extensions of the jellyfish project. The separated projects allowed us to move quickly initially but proves to be a
+bottleneck when it comes to development.
+
+By including Ocean API development with jellyfish, it creates a better synergy of DeFiChain open source development
+across concerns. Singular versioning, source of truth, documentation of entirety of defichain
+via [jellyfish.defichain.com](https://jellyfish.defichain.com).
+
+## `/apps/ocean-api`
```
Comment (fuxingloh) [inline apps/status-api/README.md:17]: @nichellekoh need you update this.

File: apps/status-api/testing/StatusApiTesting.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { StatusStubServer, RegisteredRoute } from './StatusStubServer'
+import { NestFastifyApplication } from '@nestjs/platform-fastify'
+import { InjectOptions, Response as LightMyRequestResponse } from 'light-my-request'
+
+/**
+ * LegacyApi Testing framework.
```
Comment (fuxingloh) [inline apps/status-api/testing/StatusApiTesting.ts:6]:
```suggestion
 * StatusApi Testing framework.
```

File: apps/status-api/src/providers/WhaleApiClientProvider.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { Injectable } from '@nestjs/common'
+import { SupportedNetwork } from '../pipes/NetworkValidationPipe'
+
+@Injectable()
+export class WhaleApiClientProvider {
+  private readonly clientCacheByNetwork: Map<SupportedNetwork, WhaleApiClient> = new Map()
```
Comment (fuxingloh) [inline apps/status-api/src/providers/WhaleApiClientProvider.ts:7]: We should follow `ocean-api`, `whale` structure where there is only one NETWORK type per server/instance.

Comment (fuxingloh) [reply apps/status-api/src/providers/WhaleApiClientProvider.ts:7]: So we can let the Load Balancer do the routing via, e.g. `/v0/mainnet` and `/v0/testnet`

File: apps/status-api/src/pipes/NetworkValidationPipe.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  ArgumentMetadata,
+  HttpException,
+  Injectable,
+  PipeTransform
+} from '@nestjs/common'
+
+export type SupportedNetwork = 'mainnet' | 'testnet'
+
+@Injectable()
+export class NetworkValidationPipe implements PipeTransform {
```
Comment (fuxingloh) [inline apps/status-api/src/pipes/NetworkValidationPipe.ts:11]: We don't need this, if we move to that structure.

File: apps/status-api/src/modules/WhaleApiModule.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import { Global, Injectable, Module } from '@nestjs/common'
+import { HealthIndicatorResult } from '@nestjs/terminus'
+import { WhaleApiClientProvider } from '../providers/WhaleApiClientProvider'
+import { ActuatorProbes, ProbeIndicator } from '@defichain-apps/libs/actuator'
+import { BlockchainInfo } from '@defichain/jellyfish-api-core/src/category/blockchain'
+
+@Injectable()
+export class WhaleApiProbeIndicator extends ProbeIndicator {
```
Comment (fuxingloh) [inline apps/status-api/src/modules/WhaleApiModule.ts:8]: We can also move the use the default RPC probe if we move to the other structure.

File: apps/status-api/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import { RootModule } from './modules/RootModule'
+import { FastifyAdapter, NestFastifyApplication } from '@nestjs/platform-fastify'
+import { NestFactory } from '@nestjs/core'
+import { ConfigService } from '@nestjs/config'
+
+export class RootServer {
```
Comment (fuxingloh) [inline apps/status-api/src/index.ts:6]: Use RootServer abstraction that was just merged.

**Review by fuxingloh — CHANGES_REQUESTED** [review]
@nichellekoh I recommend you create another PR with the above suggested change

**Comment by nichellekoh** [comment]
> @nichellekoh I recommend you create another PR with the above suggested change

@fuxingloh A separate PR has been created with the changes https://github.com/DeFiCh/jellyfish/pull/1292

**Pull request closed without merging**